### PR TITLE
Add Non-HTML Pages example

### PR DESCRIPTION
**Pull request by leoj3n**
Adds an example for this use case:

https://docs.astro.build/en/core-concepts/astro-pages/#non-html-pages

### CODE REVIEW

File: examples/non-html-pages/src/pages/company.json.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export async function get() {
```
Comment (FredKSchott) [inline examples/non-html-pages/src/pages/company.json.ts:1]: I'm not sure why this file is named `company.json.ts`, can we rename it to `about.json.ts` instead?

Comment (leoj3n) [reply examples/non-html-pages/src/pages/company.json.ts:1]: image


https://docs.astro.build/en/core-concepts/astro-pages/#non-html-pages

Comment (FredKSchott) [reply examples/non-html-pages/src/pages/company.json.ts:1]: doh! Mind changing in both places?

Comment (leoj3n) [reply examples/non-html-pages/src/pages/company.json.ts:1]: I think company is fine; maybe ask whoever wrote the original example.

**Review by FredKSchott — APPROVED** [review]

**Comment by FredKSchott** [comment]
BTW I cleaned up the example a bit to work in build as well as dev. When you build your site, there is no dev server running, so calling `fetch()` during build does nothing work. These generated files are best used on the frontend, while `import` and `fetchContent` are best used in your frontmatter.

https://github.com/withastro/astro/commit/5370bba0b09d7337927b6685029e48d05b1500bd

**Pull request merged**